src/connectors/paymentService.ts

```diff
@@ -490,4 +492,143 @@ export class PaymentService extends BaseService {
     }
     return parseInt(`${result[0].count}` || '0', 10)
   }
+
+  /*********************************
+   *                               *
+   *             Invoice           *
+   *                               *
+   *********************************/
+  findInvoice = async ({
+    id,
+    userId,
+    providerInvoiceId,
+    offset = 0,
+    limit = BATCH_SIZE,
+  }: {
+    id?: number
+    userId?: number
+    providerInvoiceId?: string
+    offset?: number
+    limit?: number
+  }) => {
+    let qs = this.knex('circle_invoice').select()
+
+    if (userId) {
+      qs = qs.where({ userId })
+    }
+
+    if (id) {
+      qs = qs.where({ id })
+    }
+
+    if (providerInvoiceId) {
+      qs = qs.where({ providerInvoiceId })
+    }
+
+    return qs.orderBy('created_at', 'desc').offset(offset).limit(limit)
+  }
+
+  createInvoiceWithTransactions = async ({
+    amount,
+    currency,
+    providerTxId,
+    providerInvoiceId,
+    subscriptionId,
+    userId,
+    prices,
+  }: {
+    amount: number
+    currency: string
+    providerTxId: string
+    providerInvoiceId: string
+    subscriptionId: number
+    userId: string
+    prices: CirclePrice[]
+  }) => {
+    const trx = await this.knex.transaction()
+    try {
+      // create subscription top up transaction
+      const transactionId = await trx('transaction')
+        .insert({
+          amount,
+          currency,
+          state: TRANSACTION_STATE.succeeded,
+          purpose: TRANSACTION_PURPOSE.subscription,
+
+          provider: PAYMENT_PROVIDER.stripe,
+          providerTxId,
+          senderId: undefined,
+          recipientId: userId,
+
+          targetType: undefined,
+          targetId: undefined,
+        })
+        .returning('id')
+
+      // create circle invoice
+      await trx('circle_invoice')
+        .insert({
+          userId,
+          transactionId: transactionId[0],
+          subscriptionId,
+          provider: PAYMENT_PROVIDER.stripe,
+          providerInvoiceId,
+        })
+        .returning('id')
+
+      // verify if it is OK to split the payment given the circle prices
+      for (const p of prices) {
+        if (p.currency !== currency) {
+          throw new ServerError(
+            `currency for '${providerTxId}' if not the same to '${p.providerPriceId}'`
+          )
+        }
+      }
+
+      const totalAmount = prices
+        .map((p) => Number(p.amount))
+        .reduce((p1, p2) => p1 + p2)
+      if (totalAmount !== amount) {
+        throw new ServerError(
+          `sum of plan prices '${totalAmount}' != invoice paid amount '${amount}'`
+        )
+      }
+
+      // create transactions to split the invoice payment to circles
+      const { id: entityTypeId } = await this.baseFindEntityTypeId(
+        TRANSACTION_TARGET_TYPE.circlePrice
+      )
+      for (const p of prices) {
+        const circle = await this.atomService.findFirst({
```

Comment:
I would not recommend to init atomService in another service. You can get rid of `atomService` by using:
- `this.baseFindById`  (paymentService extends baseService)
- `this.knex().select().where().first()`